Meson tries to find MKL by using 'mkl' as a package name to pkg-config, that does not work.  The real error is probably that -Dblas=mkl -Dlapack=mkl is passed to the meson build instead of -Dblas=mkl-dynamic-lp64-seq etc.  The simplest fix is to hack the meson.build script.

Author: Jakob Schiotz (Techn. Univ. Denmark) <schiotz@fysik.dtu.dk>

--- scipy/meson.build.old	2023-10-24 08:39:29.966344750 +0200
+++ scipy/meson.build	2023-10-24 09:08:24.832773698 +0200
@@ -153,6 +153,13 @@
 # https://github.com/mesonbuild/meson/issues/2835
 blas_name = get_option('blas')
 lapack_name = get_option('lapack')
+# Hack for MKL in easybuild
+if blas_name == 'mkl'
+  blas_name = 'mkl-dynamic-lp64-seq'
+endif
+if lapack_name == 'mkl'
+  lapack_name = 'mkl-dynamic-lp64-seq'
+endif
 # pkg-config uses a lower-case name while CMake uses a capitalized name, so try
 # that too to make the fallback detection with CMake work
 if blas_name == 'openblas'
